@xen-orchestra/backups/_runners/_vmRunners/_AbstractXapi.mjs

```diff
@@ -231,10 +239,13 @@ export const AbstractXapi = class AbstractXapiVmBackupRunner extends Abstract {
         ...allSettings[scheduleId],
         ...allSettings[this._vm.uuid],
       }
-      // ensure we never delete the last one for delta
-      const minRetention = this.job.mode === 'delta' ? 1 : 0
-      const retention = Math.max(settings.snapshotRetention ?? 0, minRetention)
+      const retention = settings.snapshotRetention ?? 0
       await asyncMap(getOldEntries(retention, datetimes), async datetime => {
+        // keep the last snapshot across all schedules for delta
+        // since we'll need it to compute delta for next backup
+        if (this.job.mode === 'delta' && datetime === lastSnapshotDateTime) {
```

Comment:
Why don't you compare snapshot UUIDs instead?
Wouldn't that be more precise?

Reply:
here we have either :
* a Vm snapshot with its vdi snapshot took at the same time ( at least with the same timestamp in the metadata)
* a collection of VDI snapshot took at the same time, because when user choose "purge snapshot data", we have to destroy the VM, before puring the vdi snapshot data.

All these VDI snapshot are linked to the same original VM 

I am not sure what uuid can be used there.